AtmosphereAutopilot/Modules/ProgradeThrustController.cs

```diff
@@ -31,6 +31,36 @@ public enum SpeedType
         KIAS
     }
 
+    public static class SpeedTypeMethods
+    {
+        // SpeedType classification
+        public static bool isKinematicSystem(this SpeedType t)
```

Comment:
I think if you would go one step further and not treat this as bool, the LoC would reduce. We actually already have three unit systems:

kinematic (mps, kts),
pressure (ias, kias),
sound (mach)

Old code was treating kinematic mps as lingua franca and ran all conversions through it. That is why the old code was 'switch' and not 'if' based. Can you try and separate unit system to some tangible named entity? Either another (maybe, private) enum or class hierarchy... 

Basically, my point is, current code looks a bit ugly because the pressure/kinematic condition is too prevalent in the interface and forces you to write a lot of if clauses. And you can't extend it to new systems easily.

Reply:
> current code looks a bit ugly because the pressure/kinematic condition is too prevalent in the interface and forces you to write a lot of if clauses

Yes and I already have trouble naming functions. (e.g. "sameSystemMps" is used to avoid confusion like "currentSpeedInCurrentSystem"😂)

Actually this commit is meant to be a temporary fix and I'm still testing with it (in my actual gameplay :P). Currently it doesn't work well in IAS mode when climbing/descending (changing in dyn. pressure must be taken into account for acceleration prediction) and I'm still looking forward to another refactoring (v3 perhaps).

Please ignore my commit on the FLC branch. Changes on that branch is meaningless until we settle down a solution on this branch.